src/lib/models/billing-volume.js

```diff
@@ -79,15 +80,22 @@ class BillingVolume extends Model {
 
   /**
   * The volume determined by returns algorithm
-  * @return {Number}
+  * @return {Decimal|Null}
   */
   get calculatedVolume () {
-    return this._calculatedVolume;
+    return this._calculatedVolume || null;
   }
 
   set calculatedVolume (calculatedVolume) {
-    validators.assertNullableQuantity(calculatedVolume);
-    this._calculatedVolume = isNull(calculatedVolume) ? null : parseFloat(calculatedVolume);
+    if (isNull(calculatedVolume)) {
+      this._calculatedVolume = null;
+    } else {
+      const value = new Decimal(calculatedVolume);
+      if (value.isNegative()) {
+        throw new Error(`Expected zero or positive number or decimal, got ${value.toNumber()}`);
```

Comment:
Not important. Just nitpicking 👇🏽 

<pedantry>
Consider rewording the error to `Expected zero or positive figure` purely because a number can be both positive and decimal.
</pedantry>

Reply:
Ah-ha, I meant Decimal instance, I've re-worded the message to hopefully make it clearer